plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsApplicationResource.java

```diff
@@ -27,13 +27,55 @@
 
 public class StrutsApplicationResource extends PostfixedApplicationResource {
 
+    private static final String URL_FILE_PROTOCOL = "file";
+
     private final URL url;
 
+
+    /**
+     * Generate an alternate localePath parameter string for the given URL, provided
+     * it uses a file protocol.  The alternate file localePath can provide a full
+     * path including a subcontext (e.g. Some_Path#SomeSubcontext).
+     * 
+     * @param url
+     * 
+     * @return 
+     */
+    public static String alternateFileLocalePath(URL url) {
+        final String result;
+        final String urlProtocol = url.getProtocol();
+
+        if (URL_FILE_PROTOCOL.equalsIgnoreCase(urlProtocol)) {
+            final String urlExternalForm = url.toExternalForm();
+            if (urlExternalForm != null && urlExternalForm.startsWith(urlProtocol + ':')) {
+                result = urlExternalForm.substring(urlProtocol.length() + 1);
+            } else {
+                result = url.getPath();
+            }
+        } else {
+            result = url.getPath();
+        }
+
+        return result;
+    }
+
     public StrutsApplicationResource(URL url) {
         super(url.getPath());
         this.url = url;
     }
 
+    /**
+     * Construct a StrutsApplicationResource that has the option of using 
+     * alternate file resource handling logic (for subcontexts).
+     * 
+     * @param url
+     * @param useAlternateFileResourceHandling 
+     */
+    public StrutsApplicationResource(URL url, boolean useAlternateFileResourceHandling) {
```

Comment:
Instead of a flag I would introduce another version of `StrutsApplicationResource` and create a proper instance accordingly.